src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -1519,16 +1529,26 @@ static void close_from_api(grpc_exec_ctx *exec_ctx,
                                  1);
 }
 
+typedef struct {
+  grpc_exec_ctx *exec_ctx;
+  gpr_slice *optional_drop_message;
+} cancel_stream_cb_arg;
+
 static void cancel_stream_cb(grpc_chttp2_transport_global *transport_global,
                              void *user_data,
                              grpc_chttp2_stream_global *stream_global) {
-  cancel_from_api(user_data, transport_global, stream_global,
-                  GRPC_STATUS_UNAVAILABLE);
+  cancel_stream_cb_arg *arg = user_data;
+  cancel_from_api(arg->exec_ctx, transport_global, stream_global,
```

Comment:
Fixed. Transport is obtained from transport_global using TRANSPORT_FROM_GLOBAL macro. Thanks for the review!